### PR TITLE
chore: upgrade fedimint-server to bitcoin v0.30

### PR DESCRIPTION
**Pull request by tvolk131**

### CODE REVIEW

**Review by dpc — APPROVED** [review]

File: fedimint-core/src/bitcoin_migration.rs [hunk 1]
```diff
@@ -8,6 +8,66 @@ pub fn bitcoin29_to_bitcoin30_public_key(pk: bitcoin::PublicKey) -> bitcoin30::P
         .expect("Failed to convert bitcoin v29 public key to bitcoin v30 public key")
 }
 
+pub fn bitcoin29_to_bitcoin30_secp256k1_public_key(
```
Comment (elsirion) [inline fedimint-core/src/bitcoin_migration.rs:11]: I wish we weren't just duplicating these functions everywhere. But I guess the faster we  merge all these PRs the sooner we can get rid of them.

**Review by elsirion — APPROVED** [review]

File: fedimint-core/src/bitcoin_migration.rs (hunk 1 above)
Comment (tvolk131) [reply fedimint-core/src/bitcoin_migration.rs:11]: Yeah I decided to hold off on more bitcoin v0.30 PRs until we merge the ones I have out already for this reason. Sorry for the mess! I think we've added most of the commonly needed conversion functions by now.

**Review by dpc — DISMISSED** [review]

**Review by elsirion — APPROVED** [review]

**Review by elsirion — APPROVED** [review]

**Comment by elsirion** [comment]
Needed a rebase :(

**Comment by tvolk131** [comment]
> Needed a rebase :(

Done!

**Review by dpc — APPROVED** [review]

**Pull request merged**